src/touchhandler/TouchHandler.cpp

```diff
@@ -75,15 +100,22 @@ bool TouchHandler::GetNewTouchInfo() {
   if (!info.touching) {
     gestureReleased = true;
   }
+#endif
 
   return true;
 }
 
 void TouchHandler::UpdateLvglTouchPoint() {
   if (info.touching) {
+#if defined(DRIVER_TOUCH_GESTURE)
+    // GESTURE config only generates a single event / state change
+    // so the LVGL wrapper is used to generate a successive release state update
+    lvgl.SetNewTap(info.x, info.y);
+#else
     if (!isCancelled) {
       lvgl.SetNewTouchPoint(info.x, info.y, true);
     }
+#endif
```

Comment:
While I don't have these devices to test, I think this function may be able to be simplified further. On the PineTime, a swipe immediately switches the screen and `isCancelled` is used to ignore the rest of the touch until it has been released. On the other devices the swipe event only occurs on release if I understand correctly, so parts of this could be removed.

Having preprocessor in the middle of a function isn't very nice, so depending on how different the drivers end up being, we can see if it is worth creating separate files for them or create new functions that are only included in certain builds.

Reply:
I'm not sure it would lead to a lot of code duplication. Firstly `ConvertGesture()` could be moved to Cst816s.cpp for example. Then there's not that much code left in the first place and while there are *only* two pre-processor macros, they end up touching a large portion of the code. It would also be easier to find optimizations if they were separated and freely let them diverge. For example removing `isCancelled` from `GESTURE` and perhaps `REPORT` if possible.

If the end of `UpdateLvglTouchPoints()`, which is responsible for the release event, doesn't do anything on `GESTURE`, the entire function could be replaced with a simpler one for better visibility.